pkg/virt-controller/watch/application.go

```diff
@@ -173,31 +194,58 @@ func Execute() {
 
 	app.migrationInformer = app.informerFactory.VirtualMachineInstanceMigration()
 
-	cache.WaitForCacheSync(stopChan, configMapInformer.HasSynced)
-	app.clusterConfig = virtconfig.NewClusterConfig(configMapInformer, app.kubevirtNamespace)
-
-	if app.clusterConfig.DataVolumesEnabled() {
+	if app.hasCdi {
 		app.dataVolumeInformer = app.informerFactory.DataVolume()
-		log.Log.Infof("DataVolume integration enabled")
+		log.Log.Infof("CDI detected, DataVolume integration enabled")
 	} else {
 		// Add a dummy DataVolume informer in the event datavolume support
 		// is disabled. This lets the controller continue to work without
 		// requiring a separate branching code path.
 		app.dataVolumeInformer = app.informerFactory.DummyDataVolume()
-		log.Log.Infof("DataVolume integration disabled")
+		log.Log.Infof("CDI not detected, DataVolume integration disabled")
 	}
 
 	app.initCommon()
 	app.initReplicaSet()
 	app.initVirtualMachines()
 	app.initDisruptionBudgetController()
 	app.initEvacuationController()
-	app.Run()
+	go app.Run()
+
+	select {
+	case <-app.reInitChan:
+		cancel()
+	}
+}
+
+// Detects if a config has been applied that requires
+// re-initializing virt-controller.
+func (vca *VirtControllerApp) configModificationCallback() {
+
+	vca.reInitLock.Lock()
+	defer vca.reInitLock.Unlock()
+
+	if vca.reInitTriggered {
```

Comment:
What does this do?

Reply:
keeps us from closing the channel twice (would cause crash) after we've already closed the reInit channel. Basically, there's no guarantee the process has torn down before someone could alter something that triggers teh kubevirt-config modication callback.